src/navigationValidation.js

```diff
@@ -0,0 +1,220 @@
+/*
+Validation checks to determine if app set up is ready for machine learning
+training. Panels prompt users to incrementally complete actions in preparation
+for training. Navigation is blocked if the user has not yet completed the
+action for a given panel.
+*/
+
+/*
+const panelList = [
+  { id: "selectDataset", label: "Import" },
+  { id: "dataDisplayLabel", label: "Label" },
+  { id: "dataDisplayFeatures", label: "Features" },
+  { id: "trainModel", label: "Train" },
+  { id: "generateResults", label: "Test" },
+  { id: "results", label: "Results" },
+  { id: "predict", label: "Predict" },
+  { id: "saveModel", label: "Save" },
+  { id: "modelSummary", label: "Finish" }
+];
+*/
+
+// Is a panel ready to be visited?  This determines whether a visible
+// nav button is enabled or disabled.
+export function isPanelEnabled(state, panelId) {
+
+  if (panelId === "dataDisplayLabel") {
+    if (!isDataUploaded(state)) {
```

Comment:
Alternatively, you could also replace the inner `if` with `return isDataUploaded(state)`